test/samples/eq_encoded.sol

```diff
@@ -0,0 +1,22 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.19;
```

Comment:
Could we add an executable regression test that calls eq_encoded (i.e. a vm test?)?

We should have a negative case (checks that eq_encoded fails when we want it to), and a positive test, for each of the following cases:

1. Comparing two things of different types (e.g. number and array) - negative test, always fails
2. Comparing two primitive things of the same type - numbers, addresses, booleans, fixed bytes, enums (should succeed when equal, fail when not)
3. Comparing two arrays (again check that eq_encoded succeeds when equal, fails when not)
4. Comparing two structs (same 2 tests)

Reply:
Will do.

Reply:
Addressed in e060422